pkg/appgw/ingress_rules_test.go

```diff
@@ -0,0 +1,170 @@
+package appgw
+
+import (
+	"testing"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/extensions/v1beta1"
+)
+
+func TestIngressRules(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test parsing of ingress rules")
+}
+
+var _ = Describe("Process ingress rules, listeners, and ports", func() {
+	port80 := int32(80)
+	port443 := int32(443)
+
+	expectedListener80 := frontendListenerIdentifier{
+		FrontendPort: port80,
+		HostName:     testFixturesHost,
+	}
+
+	expectedListenerAzConfigNoSSL := frontendListenerAzureConfig{
+		Protocol: "Http",
+		Secret: secretIdentifier{
+			Namespace: "",
+			Name:      "",
+		},
+		SslRedirectConfigurationName: "",
+	}
+
+	expectedListener443 := frontendListenerIdentifier{
+		FrontendPort: 443,
+		HostName:     testFixturesHost,
+	}
+
+	expectedListenerAzConfigSSL := frontendListenerAzureConfig{
+		Protocol: "Https",
+		Secret: secretIdentifier{
+			Namespace: testFixturesNamespace,
+			Name:      testFixturesNameOfSecret,
+		},
+		SslRedirectConfigurationName: agPrefix + "-" +
+			testFixturesNamespace +
+			"-" +
+			testFixturesName +
+			"-sslr",
+	}
+
+	Context("ingress rules without certificates", func() {
+		certs := getCertsTestFixture()
+		cb := makeConfigBuilderTestFixture(&certs)
```

Comment:
Just a consistency check.
Can we use `new` instead of `make` as the verb.

Reply:
Yes!  Old habit!  Will merge this and refactor all of these everywhere....
Thanks!

Reply:
Here is the follow-up PR, which renames all of these: https://github.com/Azure/application-gateway-kubernetes-ingress/pull/182